.github/workflows/main.yml

```diff
@@ -0,0 +1,70 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: NetBeans Jackpot 3.0
+
+on:
+  push:
+  pull_request:
+    # unlocked event is used as super secret restart button
+    types: [opened, synchronize, unlocked]
+
+# cancel other PR workflow run in the same head-base group if it exists (e.g. during PR syncs)
+# if this is not a PR run (no github.head_ref and github.base_ref defined), use an UID as group
+concurrency: 
+  group: ${{ github.head_ref || github.run_id }}-${{ github.base_ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  # primary build job, most other jobs use the artifact produced here
+  # artifact is only produced once in the matrix
+  base-build:
+    name: Build Jackpot 3.0
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+
+      - name: Checkout ${{ github.ref }} ( ${{ github.sha }} )
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          submodules: false
+
+      - name: Set up Ubuntu dependencies
+        run: |
+            sudo apt install ant ant-optional openjdk-11-jdk openjdk-17-jdk openjdk-21-jdk
```

Comment:
tools like `ant/mvn` are already pre-installed on the runner.

regarding java, this would be the equivalent and use the github cache which is much faster:
```yaml
      - name: Setup Java
        uses: actions/setup-java@v4
        with:
          distribution: 'temurin'
          java-version: |
            21
            17
            11
```
paths would be stored in `JAVA_HOME_11`, `JAVA_HOME_17` and `JAVA_HOME_21`, the _last_ specified JDK is used for `JAVA_HOME` and system path, which is 11 in the case above.

Reply:
I've remove the ant download. Regarding JDK, I think I'd rather avoid words like "temurin", so I've kept the installation from Ubuntu repositories. Thanks for the comment anyway!.

Reply:
For political reasons we shall ignore the benefits of the TCK! 😄

Reply:
you could have used any other vendor, I don't really care - was just an example.